jetty-core/jetty-io/src/test/java/org/eclipse/jetty/io/ContentSourceTest.java

```diff
@@ -146,16 +222,17 @@ public void run()
                     // Second demand after the first must throw.
                     assertThrows(IllegalStateException.class, () -> source.demand(this));
                 }
+                processed.countDown();
             }
         });
-        assertTrue(processed.get());
+        assertTrue(processed.await(10, TimeUnit.SECONDS));
```

Comment:
The previous use of `AtomicBoolean` made this test stricter on purpose. Using a `CountDownLatch` is not exactly the same (there was no need to change it).

Reply:
Without the latch, the test is assuming that the demand callback will be called within the scope of the demand call. That's not necessarily true in general, even if content is available.  It is definitely not true for OutputStreamContentSource, which uses a different thread to add the content after a demand call.